pkg/controller/privatelink/actuator/awsactuator/awshubactuator_test.go

```diff
@@ -0,0 +1,1494 @@
+				},
+			}),
+		),
+		AWSClientConfig: func(m *mock.MockClient) {
+			m.EXPECT().DescribeVpcEndpoints(gomock.Any()).Return(&ec2.DescribeVpcEndpointsOutput{
+				VpcEndpoints: []*ec2.VpcEndpoint{mockEndpoint},
+			}, nil)
+		},
+		expectResult: hivev1.AWSAssociatedVPC{
+			AWSPrivateLinkVPC: hivev1.AWSPrivateLinkVPC{
+				VPCID:  *mockEndpoint.VpcId,
+				Region: testRegion,
+			}},
+	}, { // The first VPC should be returned that shares the same credentials due to not being defined
+		name: "first vpc with credential null",
+		cd:   cdBuilder.Build(),
+		config: &hivev1.AWSPrivateLinkConfig{
+			CredentialsSecretRef: corev1.LocalObjectReference{Name: "credential-1"},
+			AssociatedVPCs: []hivev1.AWSAssociatedVPC{{
+				AWSPrivateLinkVPC:    hivev1.AWSPrivateLinkVPC{VPCID: "vpc-1", Region: testRegion},
+				CredentialsSecretRef: &corev1.LocalObjectReference{Name: "credential-2"},
+			}, {
+				AWSPrivateLinkVPC: hivev1.AWSPrivateLinkVPC{VPCID: "vpc-2", Region: testRegion},
+			}},
+		},
+		expectResult: hivev1.AWSAssociatedVPC{
+			AWSPrivateLinkVPC: hivev1.AWSPrivateLinkVPC{
+				VPCID:  "vpc-2",
+				Region: testRegion,
+			}},
+	}, { // The first VPC should be returned that shares the same credentials due to matching secret
+		name: "first vpc with credential nmatches",
+		cd:   cdBuilder.Build(),
+		config: &hivev1.AWSPrivateLinkConfig{
+			CredentialsSecretRef: corev1.LocalObjectReference{Name: "credential-1"},
+			AssociatedVPCs: []hivev1.AWSAssociatedVPC{{
+				AWSPrivateLinkVPC:    hivev1.AWSPrivateLinkVPC{VPCID: "vpc-1", Region: testRegion},
+				CredentialsSecretRef: &corev1.LocalObjectReference{Name: "credential-2"},
+			}, {
+				AWSPrivateLinkVPC:    hivev1.AWSPrivateLinkVPC{VPCID: "vpc-2", Region: testRegion},
+				CredentialsSecretRef: &corev1.LocalObjectReference{Name: "credential-1"},
+			}},
+		},
+		expectResult: hivev1.AWSAssociatedVPC{
+			AWSPrivateLinkVPC: hivev1.AWSPrivateLinkVPC{
+				VPCID:  "vpc-2",
+				Region: testRegion,
+			},
+			CredentialsSecretRef: &corev1.LocalObjectReference{Name: "credential-1"},
+		},
+	}, { // There should be an error if there are no associatedVPCS configured
+		name:        "no associatedVPCS configured",
+		cd:          cdBuilder.Build(),
+		config:      &hivev1.AWSPrivateLinkConfig{},
+		expectError: "unable to find an associatedVPC that uses the primary AWS PrivateLink credentials",
+	}, { // There should be an error if no suitable associatedVPCS were configured
+		name: "no suitable associatedVPCS configured",
+		cd:   cdBuilder.Build(),
+		config: &hivev1.AWSPrivateLinkConfig{
+			CredentialsSecretRef: corev1.LocalObjectReference{Name: "credential-1"},
+			AssociatedVPCs: []hivev1.AWSAssociatedVPC{{
+				AWSPrivateLinkVPC:    hivev1.AWSPrivateLinkVPC{VPCID: "vpc-1", Region: testRegion},
+				CredentialsSecretRef: &corev1.LocalObjectReference{Name: "credential-2"},
+			}},
+		},
+		expectError: "unable to find an associatedVPC that uses the primary AWS PrivateLink credentials",
+	}}
```

Comment:
I purposely skipped test because I found it is an impossible situation. getAssociatedVPCs can only return an [error](https://github.com/openshift/hive/blob/d6f2766aef12b836a75c70e06ffdc2d99246f12e/pkg/controller/privatelink/actuator/awsactuator/awshubactuator.go#L573) when VPCEndpointID is set.

Reply:
Mm, I see. An "unreachable" comment on the code side, then?

Reply:
[Done](https://github.com/openshift/hive/compare/64bc793e2968e4d36b639cf8768518c6fc0992db..351b11a3f68ee550829b98153ea96b39d6604ab8)